src/testing/basic_test.rs

```diff
@@ -134,6 +127,32 @@ mod tests {
         let mut rres_msgs: Vec<ResponseMessage> = Vec::new();
         let _validated_state = Arc::new(TestValidatedState::default());
 
+        let arc_rwlock_global_state = Arc::new(RwLock::new(global_state));
+        let arc_rwlock_global_state_clone = arc_rwlock_global_state.clone();
+        let handle = async_spawn(async move {
```

Comment:
Question: Curious why the `handle` construction is moved up here. Can't it just stay where it was?

Reply:
Because now the channels aren't actually active until we spawn a `BuilderState`, so we want the handle to be spawned before we start sending something to it.